### PR TITLE
feat(inference): add posterior predictive and science output utilities

### PR DESCRIPTION
**Pull request by TobiBu**
## Summary
- add posterior predictive sampling utilities for VI outputs
- add predictive summary maps (mean/std/p16/p50/p84)
- add residual/chi2 products and masked scalar metrics helpers
- export utilities in rubix.inference
- add focused unit tests and workflow docs

## Validation
- pre-commit passed on changed files
- compileall passed for module and tests

### CODE REVIEW

File: rubix/inference/posterior_predictive.py [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+        cubes.append(
+            forward(
+                pipeline=pipeline,
+                params=sampled_constrained,
+                static_data=static_data,
+                noise_key=noise_key,
+            )
+        )
+
+    return jnp.stack(cubes, axis=0)
```
[the remark this answers is not in the document]
Comment (TobiBu) [reply rubix/inference/posterior_predictive.py]: @copilot apply changes based on [this feedback](https://github.com/AstroAI-Lab/rubix/pull/218#discussion_r3116290841)

**Comment by TobiBu** [comment]
@copilot can you try again to apply the changes based on the feedback from this PR?
